src/stashcache

```diff
@@ -1,198 +1,85 @@
-
-def manage_xrootd(signum, frame):
-    '''Signal handler for the XRootD service'''
-    command = _accepted_signals[signum]
-    htcondor.log(htcondor.LogLevel.Always,
-                 "Received signal %s: Running '%s' on the XRootD server " % (signum, command))
-
-    rc, _, _ = xrootd_service_command(command)
-    if rc != 0:
-        kill_xrootd()
-        # Restart service if user requested it
-        if signum == signal.SIGHUP:
-            start_xrootd()
-    if signum == signal.SIGQUIT or signum == signal.SIGTERM:
-        sys.exit(0)
-
-def start_xrootd():
-    '''Starts the XRootD init service'''
-    rc, stdout, _ = xrootd_service_command('start')
-    if rc != 0:
-        htcondor.log(htcondor.LogLevel.Error, 'ERROR: Could not start XRootD service: %s' % stdout)
-    else:
-        try:
-            pid = get_xrootd_pid()
-        except AttributeError:
-            pid = 'unknown'
-        htcondor.log(htcondor.LogLevel.Always, 'XRootD service running (PID: %s)' % pid)
-
-def kill_xrootd():
-    '''Sends SIGKILL to an unresponsive XRootD service'''
-    try:
-        pid = get_xrootd_pid()
-        os.kill(int(pid), signal.SIGKILL)
-    except (AttributeError, OSError):
-        pass # xrootd already dead
-
-def get_xrootd_pid():
-    '''Finds the xrootd pid via `service xrootd status`'''
-    _, stdout, _ = xrootd_service_command('status')
+def heart_beat(hostname, cache_path='/stash', one_shot=False):
+    '''Send heartbeats and cache ads to the condor_master'''
     try:
-        return re.search(r'(?:pid|PID)[:\s\(]*(\d+)', stdout).group(1)
-    except AttributeError:
-        htcondor.log(htcondor.LogLevel.Error, 'ERROR: Could not find XRootD service PID')
-        raise
+        central_coll_url = htcondor.param['OSG_COLLECTOR_HOST']
+    except KeyError:
+        central_coll_url = 'collector1.opensciencegrid.org:9619,collector2.opensciencegrid.org:9619'
+    logging.debug('Using %s as the central collector', central_coll_url)
+    central_coll = htcondor.Collector(central_coll_url)
 
-def xrootd_service_command(cmd):
-    '''Accepts an init command and passes it to the XRootD init script'''
-    service_name = 'xrootd'
-    try:
-        # To support systemd, we need to specify which configuration xrootd should use
-        rc, _, _ = run_command(['systemctl', '--version'])
-        for config in ('cache', 'origin'):
-            if os.path.exists('/etc/xrootd/xrootd-stashcache-%s-server.cfg' % config):
-                service_name = 'xrootd@stashcache-%s-server' % config
-                break
-        if service_name is 'xrootd':
-            htcondor.log(htcondor.LogLevel.Error, 'ERROR: Could not %s XRootD service + ' \
-                         'due to missing cache and origin configurations.' % cmd)
-            sys.exit(1)
-    except OSError, e:
-        # systemctl isn't installed, assume sysvinit
-        if e.errno == errno.ENOENT:
-            pass
+    xrootd_url = 'root://' + hostname
+    while True:
```

Comment:
Maybe I don't see it, but does this just run forever?  No sleep or wait?

Reply:
The changes in indent made the diff rather messy. There's a sleep here:
https://github.com/opensciencegrid/StashCache-Daemon/blob/80f901026a/src/stashcache#L49